### PR TITLE
Parameterize emulator config

### PR DESCRIPTION
**Pull request by fletcherm**
This addresses #8 

I decided not to try adding tests for this due to the static nature of things. IMO breaking this up a bit when I go to improve config file loading in general (for #1, and Paladine's currently open pull request). Then tests will be significantly easier to add.

Eager for someone else to manually test it and see where it does & doesn't hold up :)

### CODE REVIEW

File: MBBSEmu/Program.cs [hunk 1]
```diff
@@ -96,6 +118,18 @@ static void Main(string[] args)
                     }
                 }
 
+                if (sSettingsFile == null)
```
Comment (fletcherm) [inline MBBSEmu/Program.cs:121]: @enusbaum Question about this `if` block and how C# default parameters work.

Can all this be condensed down to simply `ServiceResolver.Create(sSettingsFile)`?

I'm hoping that works, but my fear is that there's some subtle difference between the programmer explicitly setting `null`, vs. the variable implicitly being `null`, and how the default parameter in the `Create` function behaves.

Please tell me I'm overthinking this and we can delete some code here...

Comment (enusbaum) [reply MBBSEmu/Program.cs:121]: You could define default file name in this class as a constant and do something like `ServiceResolver.Create(sSettingsFile ?? DefaultFile)` (?? being an if-null-then) :)

Comment (fletcherm) [reply MBBSEmu/Program.cs]: Thanks - in learning more about `??` I also learned about `??=`. Ended up using that as IMO it removed some subtleties hiding in the syntax.

**Comment by fletcherm** [comment]
@enusbaum Looks for your feedback here - the latest build failed.

First "Details" link for "Build #43 failed" gives me an error message that is meaningless to me

Second "Details" link gives me a 401

Are you able to make more sense of this?

**Pull request merged**